src/EditorFeatures/CSharpTest/ExtractClass/ExtractClassTests.cs

```diff
@@ -197,6 +197,69 @@ int Method2()
             }.RunAsync();
         }
 
+        [Fact]
+        public async Task TestRecord()
+        {
+            var input = """
+                record R(string S)
+                {
+                    void $$M()
+                    {
+                    }
+                }
+                """;
+
+            var expected1 = """
+                record R(string S) : MyBase
+                {
+                }
+                """;
+
+            var expected2 = """
+                internal record MyBase
+                {
+                    void M()
+                    {
+                    }
+                }
+
+                """;
+
+            await new Test
+            {
+                TestCode = input,
+                LanguageVersion = LanguageVersion.CSharp9,
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                FixedState =
+                {
+                    Sources =
+                    {
+                        expected1,
+                        expected2,
+                    }
+                },
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestRecord2()
+        {
+            // https://github.com/dotnet/roslyn/issues/62415 to make this scenario work
+            var input = """
+                record R(string $$S)
+                {
+                }
+                """;
+
+            await new Test
+            {
+                TestCode = input,
+                FixedCode = input,
+                LanguageVersion = LanguageVersion.CSharp9,
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+            }.RunAsync();
+        }
+
```

Comment:
Test that we don't offer for `record struct`?

Would also be good to test

```
record R
{
	public string S {get;set;}
}
```
and also
```
record R(string S)
{
	public string S {get;set;} = S;
}
```

Reply:
Added tests. The second example will definitely still produce bad code, I'll enable that in my second pass